python/pyarrow/parquet/__init__.py

```diff
@@ -3101,6 +3101,26 @@ def file_visitor(written_file):
     >>> pq.ParquetDataset('dataset_name_4/', use_legacy_dataset=False).files
     ['dataset_name_4/...-0.parquet']
     """
+    # Check for conflicting kewords
+    msg_conflicting = (
+        "Arguments '{0}' and '{1}' can not be passed together. "
+        "'{0}' is supported with the new dataset implementation "
+        "and '{1}' with the legacy implementation."
+    )
```

Comment:
Would it make sense to do something like switch the message based on use_legacy_dataset so that the error will be something like `"{0} is not supported by use_legacy_dataset=False, use {1}"` vs. `"{1} is not supported by use_legacy_dataset=True, use {0}"`?

Reply:
Sure 👍 Will do